### PR TITLE
Add mesh example of 2D quadrilateral-based icon

### PR DESCRIPTION
**Pull request by Yu-Rou-Weng**
- For-Discussion PR: mesh example of 2D Quadrilateral-based sciwork icon

- In sample_mesh.py, add ```make_sciwork()``` to show interesting mesh example

sciwork_icon

### CODE REVIEW

File: modmesh/app/sample_mesh.py [hunk 1]
```diff
@@ -89,6 +89,18 @@ def help_3dmix(set_command=False):
     if set_command:
         view.mgr.pycon.command = cmd.strip()
 
+def help_sciwork(set_command=False):
```
Comment (yungyuc) [inline modmesh/app/sample_mesh.py]: Quite nice.  But can we use "solvcon", or "modmesh"?  This project is not part of sciwork, while it happens to participate in sciwork activities.

**Review by yungyuc — CHANGES_REQUESTED** [review]
Need to change:

- [x] Use "solvcon" or "modmesh" for the text, not "sciwork".

Is it OK to have both 2D and 3D versions?  (This is an open question).

The lower-left in "R" has point touch, which does not make sense in a mesh.  But once we change the text to "solvcon" and/or "modmesh", "R" is not there anymore, so it's OK.

**Comment by yungyuc** [comment]
I think the PR is not in a work in progress "WIP" state.  Please change it by hitting "ready for review".

File: modmesh/app/sample_mesh.py (hunk 1 above)
Comment (Yu-Rou-Weng) [reply modmesh/app/sample_mesh.py]: Sure! I can swift to use "solvcon" or "modmesh". I think I'll first try to use 2D to represent it and then think about how to optimize for using 3D.

**Comment by Yu-Rou-Weng** [comment]
@yungyuc Hi yyc, I've changed the text of 2D icon to "solvcon", I placed the first letter s slightly higher than the other letters to avoid distortion of the letters.

solvcon圖片

**Review by yungyuc — CHANGES_REQUESTED** [review]
Please fix the linter error in the CI: https://github.com/solvcon/modmesh/actions/runs/8220013667/job/22509416289?pr=285 .

There may be other CI failures.  Should you have any questions please go to sciwork solvcon channel.

**Comment by Yu-Rou-Weng** [comment]
@yungyuc I have make sure all CI task run well on my fork, thanks!

**Review by yungyuc — APPROVED** [review]
LGTM

File: modmesh/app/sample_mesh.py [hunk 2]
```diff
@@ -222,6 +236,112 @@ def make_3dmix():
...
+    mh.cltpn.ndarray[:] = [
+           Q, Q, Q, Q, Q, Q, Q, Q, Q, Q,
+           Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q,  # 0-20
+           Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q,  # 21-31
+           Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q,  # 32-45
+           Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q, Q,  # 46-61
+           Q, Q, Q  # 62-64
+        ]
+    mh.clnds.ndarray[:, :5] = [
+            (4, 0, 1, 30, 29),
```
Comment (tigercosmos) [inline modmesh/app/sample_mesh.py:273]: @yungyuc  isn't it better to put this in a data file rather than hard-coded in a py file?

**Pull request merged**